### PR TITLE
fix(docs): restore behavior of description term links

### LINKED ISSUE #9810: Unable to recognize definition term hyperlinks in definition lists
### Summary

After https://github.com/mdn/yari/pull/8413 landed all the definition terms got same formatting. Now it is not possible to recognize hyperlinks for these terms.

### URL

https://developer.mozilla.org/en-US/docs/Web/CSS/border

### Expected behavior

The definition term hyperlinks be recognizable.
This is how it looked prior to the changes:
https://web.archive.org/web/20231011150428/https://developer.mozilla.org/en-US/docs/Web/CSS/border#values
> 

Not asking to completely revert the changes but to fix/improve the implementation.

### Actual behavior

> 

cc/ @estelle

### PR DESCRIPTION
**Pull request by caugner**
## Summary

Fixes #9810.

### Problem

https://github.com/mdn/yari/pull/8413 introduced a regression, making it unable to identify description terms (`<dt>`) that link to another page.

### Solution

Restore the style of those links, excluding anchor links.

Also makes the hashtag of the anchor links interactive.

---

### Before

image

### After

image

---

## How did you test this change?

Ran `yarn dev` and opened http://localhost:3000/en-US/docs/Web/CSS/border#values locally.

### CODE REVIEW

**Comment by caugner** [comment]
@OnkarRuikar Would you be interested in reviewing this PR?

File: client/src/document/index.scss [hunk 1]
```diff
@@ -148,7 +148,6 @@
       a {
         &:link,
         &:visited {
-          color: inherit;
           text-decoration: none;
```
Comment (OnkarRuikar) [inline client/src/document/index.scss:151]:
```suggestion
          text-decoration: underline;
        }

        &:hover,
        &:focus {
          text-decoration: none;
        }

        &[href^="#"] {
          color: inherit;
          position: relative;

          &:link,
          &:visited {
            text-decoration: none;
          }

          &:hover,
          &:focus {
            text-decoration: underline;
          }

          &::before {
            color: var(--text-inactive);
            content: "#";
            display: inline-block;
            font-size: 0.7em;
            line-height: 1;
            margin-left: -0.8em;
            text-decoration: none;
            visibility: hidden;
            width: 0.8em;
          }

          &:hover::before {
            visibility: visible;
```
1. For underline, hyperlinks and page anchors have opposite behavior. Hyperlinks have underlines and on hover it is removed. For page anchors underlines are not visible but are shown on hover.
2. The `#` on page anchors is not interactible with `position: absolute`. It works well with same pseudo-element implementation of H1-6 headers.

Comment (OnkarRuikar) [inline client/src/document/index.scss:151]:
```suggestion
          text-decoration: underline;
        }

        &:hover,
        &:focus {
          text-decoration: none;
        }

        &[href^="#"] {
          color: inherit;
          position: relative;

          &:link,
          &:visited {
            text-decoration: none;
          }

          &:hover,
          &:focus {
            text-decoration: underline;
          }

          &::before {
            color: var(--text-inactive);
            content: "#";
            display: inline-block;
            font-size: 0.7em;
            line-height: 1;
            margin-left: -0.8em;
            text-decoration: none;
            visibility: hidden;
            width: 0.8em;
          }

          &:hover::before {
            visibility: visible;
```
1. For underline, hyperlinks and page anchors have opposite behavior. Hyperlinks have underlines and on hover it is removed. For page anchors underlines are not visible but are shown on hover.
2. The `#` on page anchors is not interactible with `position: absolute`. It works well with same pseudo-element implementation of H1-6 headers.

Comment (OnkarRuikar) [reply client/src/document/index.scss:151]: I selected lines till 172 for the review but it shows review only for 151. :confused:

**Comment by OnkarRuikar** [comment]
This how the suggestion works:

https://github.com/mdn/yari/assets/87750369/f5646c41-ef8f-4b22-8844-14c725466f22

**Comment by caugner** [comment]
@OnkarRuikar Thank you for catching those issues! 🙌

File: client/src/document/index.scss [hunk 2]
```diff
@@ -146,30 +146,31 @@
       margin-top: 2rem;
 
       a {
```
Comment (OnkarRuikar) [inline client/src/document/index.scss:148]:
```suggestion
      a[href^="#"] {
```

File: client/src/document/index.scss [hunk 3]
```diff
@@ -146,30 +146,31 @@
...
-        &:link,
-        &:visited {
+        &[href^="#"] {
```
Comment (OnkarRuikar) [inline client/src/document/index.scss:149]:
```suggestion
```

**Review by OnkarRuikar — CHANGES_REQUESTED** [review]
The code is much more concise now. A tiny simplification and it will be perfect.

**Review by OnkarRuikar — APPROVED** [review]

**Comment by caugner** [comment]
Thank you for the thorough reviews @OnkarRuikar! 🙌

**Pull request merged**